### ISSUE TITLE
Flashing screen when holding a loading chip

### ISSUE THREAD

**Issue by El-Mico** [Rusketh/ExpAdv2#317]
Sometimes when a player grabs a chip while it's still loading, some players screens will flash with different colors (seems to be linked to the player's physgun color, and the phys-beam itself).

**Comment by bigdogmat** [comment]
This is intended while the script is compiling

**Comment by Rusketh** [comment]
shouldn't be.

**Comment by bigdogmat** [comment]
As far as I remember using ExpAdv2, whenever you would pick up a compiling chip it would flash bright yellow-green colors on your screen. I always just thought this was intentional.

**Comment by Codingale** [comment]
This, I figured it was an chip I found that caused it but after a server restart I noticed it does it regardless.

Here's a video of it happening randomly I found, though the audio was massively corrupted so I replaced it since YouTube doesn't let me remove it:
 https://www.youtube.com/watch?v=grds5AcILdk

**Comment by bigdogmat** [comment]
Your script compiles almost instantly, that's why you're only seeing a flash of it.

**Comment by Codingale** [comment]
Yeah, I know, found out just having a back-tick (`) by it's self causes the compiler to completely fail and keep loading indefinitely.

**Comment by bigdogmat** [comment]
So I found out why this happens, turns out it has to do with the halo outline drawn by the physgun and stencils. It can easily be fixed with `halo.RenderedEntity` and a check to see if that is equal to the entity in the draw function.

**Comment by Codingale** [comment]
Oh wow, completely forgot about this.

It also happens if you hold C and mouse over it. My video is a bit bad and I know it's due to the fact I'm comping it near instantly.

Another thing, if you just put a grave accent / back tick (`) you can completely halt the compiler and abuse this for as long as you want as it never complies, but hey at least it doesn't freeze the server whilst doing this.
